package-builder/build-manifest.nix

```diff
@@ -27,13 +27,22 @@ let
   install-prefix-contents = /. + install-prefix;
   buildScript-contents = /. + buildScript;
   buildCache-tar-contents = if (buildCache == null) then null else (/. + buildCache);
-  dollar_out_error_and_exit = ''
-    echo "❌ ERROR: Build command did not copy outputs to '\$out'." 1>&2
-    echo "- copy a single file with 'mkdir -p \$out/bin && cp file \$out/bin'" 1>&2
-    echo "- copy a bin directory with 'mkdir \$out && cp -r bin \$out'" 1>&2
-    echo "- copy multiple files with 'mkdir -p \$out/bin && cp bin/* \$out/bin'" 1>&2
-    echo "- copy files from an Autotools project with 'make install PREFIX=\$out'" 1>&2
-    exit 1
+
+  dollar_out_bin_copy_hints = ''
+    echo "  - copy a single file with 'mkdir -p \$out/bin && cp file \$out/bin'" 1>&2
+    echo "  - copy a bin directory with 'mkdir \$out && cp -r bin \$out'" 1>&2
+    echo "  - copy multiple files with 'mkdir -p \$out/bin && cp bin/* \$out/bin'" 1>&2
+    echo "  - copy files from an Autotools project with 'make install PREFIX=\$out'" 1>&2
+  '';
+  dollar_out_error = ''
+    echo "❌  ERROR: Build command did not copy outputs to '\$out'." 1>&2
+    ${dollar_out_bin_copy_hints}
+  '';
+  dollar_out_no_bin_warning = ''
+    echo "⚠️  WARNING: No executables found in '\$out/bin'." 1>&2
+    echo "Only executables in '\$out/bin' will be available on the PATH." 1>&2
```

Comment:
nit: strictly speaking we also add `$out/sbin` to PATH

Reply:
right (and i didnt consistently check for either as well below), does it make sense to keep repeating `'$out/bin' or $out/sbin` everywhere? to me that seems to be a bit repetative, and adding to sbin isnt _that_ common, given that sbin is for binaries that require root privileges 🤔